File: src/main/java/com/giant/module/employee/entity/Team.java
package com.giant.module.employee.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Table(
        name = "team",
        schema = "employee",
        indexes = {
                @Index(name = "ix_team_department", columnList = "department_id")
        }
)
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class Team {
    @Id
    @Column(name = "id", nullable = false)
    private Long teamId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(
            name = "department_id",
            nullable = false,
            foreignKey = @ForeignKey(name = "fk_team_department")
    )
    private Department department;

    @Column(
            name = "team_name",
            nullable = false,
            unique = true
    )
    private String teamName;

    @OneToMany(mappedBy = "team", fetch = FetchType.LAZY)
    private List<EmployeeProfile> employeeProfiles;
}

File: src/main/kotlin/com/giant/module/employee/entity/Team.kt
package com.giant.module.employee.entity

import jakarta.persistence.*

@Entity
@Table(name = "team", schema = "employee", indexes = [Index(name = "ix_team_department", columnList = "department_id")])
class Team(
    @Column(name = "team_code", nullable = false, unique = true)
    var teamCode: String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "department_id", nullable = false, foreignKey = ForeignKey(name = "fk_team_department"))
    var department: Department,

    @Column(name = "team_name", nullable = false, unique = true)
    var teamName: String,
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    val teamId: Long? = null

    @OneToMany(mappedBy = "team", fetch = FetchType.LAZY)
    private var _employeeProfiles: MutableList<EmployeeProfile> = mutableListOf()

    val employeeProfiles: List<EmployeeProfile>
        get() = _employeeProfiles.toList()

    companion object {
        fun create(
            teamCode: String,
            department: Department,
            teamName: String
        ): Team {
            return Team(teamCode, department, teamName)
        }
    }
}
